src/main/kotlin/com/github/vol0n/utbotcppclion/actions/GenerateForPredicateAction.kt

```diff
@@ -0,0 +1,246 @@
+        }
+
+        private fun intBoundsBySize(size: Int, signed: Boolean): Pair<BigInteger, BigInteger> {
+            if (!signed) {
+                return Pair((0).toBigInteger(), (2).toBigInteger().pow(size).dec())
+            }
+            return Pair((2).toBigInteger().pow(size-1).unaryMinus(), (2).toBigInteger().pow(size-1).dec())
+        }
+
+        private val validationTypeName = mapOf(
+            ValidationType.INT8_T to "int8_t",
+            ValidationType.INT16_T to "int16_t",
+            ValidationType.INT32_T to "int32_t",
+            ValidationType.INT64_T to "int64_t",
+            ValidationType.UINT8_T to "uint8_t",
+            ValidationType.UINT16_T to "uint16_t",
+            ValidationType.UINT32_T to "uint32_t",
+            ValidationType.UINT64_T to "uint64_t"
+        )
+
+        private val integerBounds = mapOf(
+            ValidationType.INT8_T to intBoundsBySize(8, false),
+            ValidationType.INT16_T to intBoundsBySize(16, false),
+            ValidationType.INT32_T to intBoundsBySize(32, false),
+            ValidationType.INT64_T to intBoundsBySize(64, false),
+            ValidationType.UINT8_T to intBoundsBySize(8, true),
+            ValidationType.UINT16_T to intBoundsBySize(16, true),
+            ValidationType.UINT32_T to intBoundsBySize(32, true),
+            ValidationType.UINT64_T to intBoundsBySize(64, true)
+        )
+
+        private fun intValidationFunc(validationType: ValidationType): (String) -> String?  {
+            return fun(value: String): String? {
+                return if ("""^-?(([1-9][0-9]*)|0)$""".toRegex().matches(value)) {
+                    if (isIntegerInBounds(value, integerBounds[validationType]?.first, integerBounds[validationType]?.second)) {
+                        null
+                    } else {
+                        "Value does not fit into C  ${validationTypeName[validationType]} type"
+                    }
+                } else {
+                    "Value is not an integer"
+                }
+            }
+        }
+
+        val returnValueValidators = mapOf(
+            ValidationType.INT8_T to intValidationFunc(ValidationType.INT8_T),
+            ValidationType.INT16_T to intValidationFunc(ValidationType.INT16_T),
+            ValidationType.INT32_T to intValidationFunc(ValidationType.INT32_T),
+            ValidationType.INT64_T to intValidationFunc(ValidationType.INT64_T),
+            ValidationType.UINT8_T to intValidationFunc(ValidationType.UINT8_T),
+            ValidationType.UINT16_T to intValidationFunc(ValidationType.UINT16_T),
+            ValidationType.UINT32_T to intValidationFunc(ValidationType.UINT32_T),
+            ValidationType.UINT64_T to intValidationFunc(ValidationType.UINT64_T),
+            ValidationType.CHAR to fun(value: String): String? {
+            if (value.length == 1) {
+                return null
+            } else {
+                val escapeSequences = listOf(
```

Comment:
Please add a comment on why escape sequences are used, and how the user can interact with them